### ISSUE TITLE
Bugs

### ISSUE THREAD

**Issue by alanOracle** [JamesLMilner/terra-draw#37]
## When removing Terra Draw in mapbox/maplibre
Object keys should be used here since we are using an array.

https://github.com/JamesLMilner/terra-draw/blob/65ba076741e610a1b9c9eaa6baa5d71d4a1a85b8/src/adapters/mapbox-gl.adapter.ts#L35-L45

Line 35 could be...
`["point", "linestring", "polygon"].forEach((geometryKey) => {`
and 41
`if (geometryKey === "polygon") {`

## Style property assignment
2 things here, first the assignment is not being done correctly here
https://github.com/JamesLMilner/terra-draw/blob/65ba076741e610a1b9c9eaa6baa5d71d4a1a85b8/src/modes/select/select.mode.ts#L611
Could be...
`styles.pointOutlineWidth = this.styles.selectionPointOutlineWidth || 2;`

And second, what if the user wants to set the midpoint outline width to 0, then this type of evaluation would interpret that the user hast set anything, a workaround to this right now is to set the value as '0' instead of just 0 so that JavaScript interprets its a string.

## Map draggability when in select mode
Currently if you have something selected  you CAN NOT move the underlying map. Basically this is because `setDraggability` is set to false when `onDragStart` is being called. I think it is better to set draggability to false only when is strictly necessary (we know for sure the user is about to perform some modification to a feature). 

To achieve that we can remove this
https://github.com/JamesLMilner/terra-draw/blob/65ba076741e610a1b9c9eaa6baa5d71d4a1a85b8/src/modes/select/select.mode.ts#L471

And we would need to pass in some way the setDraggability function to each of the behaviors
- drag-coordinate
- drag-feature
- rotate-feature
- scale-feature

All of them have an event as a parameter so maybe we could add it to this interface
https://github.com/JamesLMilner/terra-draw/blob/65ba076741e610a1b9c9eaa6baa5d71d4a1a85b8/src/common.ts#L28-L35

like `setDraggability?: (enabled: boolean) => void;` 
which would be filled in this method of the base adapter

https://github.com/JamesLMilner/terra-draw/blob/65ba076741e610a1b9c9eaa6baa5d71d4a1a85b8/src/adapters/common/base.adapter.ts#L297-L319

like `const setDraggability = this.setDraggability.bind(this);`

Then on each of the behaviors we would add this code
```
                if (event.setDraggability) {
			event.setDraggability(false);
		}
```

Between the lines 53 and 55
https://github.com/JamesLMilner/terra-draw/blob/65ba076741e610a1b9c9eaa6baa5d71d4a1a85b8/src/modes/select/behaviors/drag-feature.behavior.ts#L49-L55

Between the lines 66 and 68
https://github.com/JamesLMilner/terra-draw/blob/65ba076741e610a1b9c9eaa6baa5d71d4a1a85b8/src/modes/select/behaviors/drag-coordinate.behavior.ts#L63-L69

Between the lines 34 and 36
https://github.com/JamesLMilner/terra-draw/blob/65ba076741e610a1b9c9eaa6baa5d71d4a1a85b8/src/modes/select/behaviors/rotate-feature.behavior.ts#L31-L36

Again between the lines 34 and 36
https://github.com/JamesLMilner/terra-draw/blob/65ba076741e610a1b9c9eaa6baa5d71d4a1a85b8/src/modes/select/behaviors/scale-feature.behavior.ts#L31-L36

This way the user could move the map freely even when has something selected, and only would prevent so if the user actually wants to do something with the feature.

**Comment by JamesLMilner** [comment]
Hi @alanOracle - thanks for these! I have fixed the first two issues `0.0.1-alpha.35`. The last issue is slightly more complicated but I have an idea on how to better handle it!

**Comment by JamesLMilner** [comment]
@alanOracle I will close this as the only specific issue left is the Select mode draggability which is a slightly bigger bit of owrk. I've added it as https://github.com/JamesLMilner/terra-draw/issues/42
